client/src/PublicRequest.ts

```diff
@@ -187,32 +187,52 @@ export type SignBtcTransactionRequest
 
 export type SignSwapRequest = SimpleRequest & {
     swapId: string,
-    fund:
+    fund: (
         {type: 'NIM'}
-            & Omit<TransactionInfo,
-                'senderType'
-                | 'recipient'
-                | 'recipientType'
-                | 'recipientLabel'
-                | 'validityStartHeight'
-                | 'data'
-                | 'flags'
-            >
-            & { senderLabel: string }
-        | {type: 'BTC', keyPaths: string[], value: number, fee: number },
-    redeem:
+        & Omit<TransactionInfo,
+            | 'recipient' // Only known in second step (in swap-iframe), derived from htlcData
+            | 'recipientType' // Must be HTLC
+            | 'recipientLabel' // Not used
+            | 'data' // Only known in second step (in swap-iframe)
+            | 'flags' // Must be CONTRACT_CREATION
+        >
+        & { senderLabel: string }
+    ) | (
+        {type: 'BTC'}
+        & Omit<BitcoinTransactionInfo,
```

Comment:
The fee is not validated in the Keyguard which is not ideal.

Reply:
How should we validate the fee?